core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java

```diff
@@ -172,16 +174,54 @@ public void removeBlock(long blockId) {
   }
 
   /**
-   * @return generated {@link WorkerInfo} for this worker
+   * Gets the selected field information for this worker.
+   *
+   * @param fieldRange the client selected fields
+   * @param isLiveWorker the worker is live or not
+   * @return generated worker information
    */
-  public WorkerInfo generateClientWorkerInfo() {
-    return new WorkerInfo()
-        .setId(mId)
-        .setAddress(mWorkerAddress)
-        .setLastContactSec(
-            (int) ((CommonUtils.getCurrentMs() - mLastUpdatedTimeMs) / Constants.SECOND_MS))
-        .setState("In Service").setCapacityBytes(mCapacityBytes).setUsedBytes(mUsedBytes)
-        .setStartTimeMs(mStartTimeMs);
+  public WorkerInfo generateWorkerInfo(Set<WorkerInfoField> fieldRange, boolean isLiveWorker) {
+    WorkerInfo info = new WorkerInfo();
+    for (WorkerInfoField field : fieldRange != null ? fieldRange
+        : Arrays.asList(WorkerInfoField.values())) {
+      switch (field) {
+        case ADDRESS:
+          info.setAddress(mWorkerAddress);
+          break;
+        case CAPACITY_BYTES:
+          info.setCapacityBytes(mCapacityBytes);
+          break;
+        case CAPACITY_BYTES_ON_TIERS:
+          info.setCapacityBytesOnTiers(mTotalBytesOnTiers);
+          break;
+        case ID:
+          info.setId(mId);
+          break;
+        case LAST_CONTACT_SEC:
+          info.setLastContactSec(
+              (int) ((CommonUtils.getCurrentMs() - mLastUpdatedTimeMs) / Constants.SECOND_MS));
+          break;
+        case START_TIME_MS:
+          info.setStartTimeMs(mStartTimeMs);
+          break;
+        case STATE:
+          if (isLiveWorker) {
+            info.setState("In Service");
```

Comment:
define string constant of "In Service" and "Out of service" rather than hard code them here.

Reply:
Changed to 
```java
  private static final String LIVE_WORKER_STATE = "In Service";
  private static final String LOST_WORKER_STATE = "Out of Service";
```
Thanks